bdb/drivers/mssql.go

```diff
@@ -0,0 +1,367 @@
+		columns = append(columns, column)
+	}
+
+	if err = rows.Err(); err != nil {
+		return nil, err
+	}
+
+	pkey.Columns = columns
+
+	return pkey, nil
+}
+
+// ForeignKeyInfo retrieves the foreign keys for a given table name.
+func (m *MSSQLDriver) ForeignKeyInfo(schema, tableName string) ([]bdb.ForeignKey, error) {
+	var fkeys []bdb.ForeignKey
+
+	query := `
+	SELECT 
+		ccu.constraint_name
+		,ccu.table_name AS local_table
+		,ccu.column_name AS local_column
+		,kcu.table_name AS foreign_table
+		,kcu.column_name AS foreign_column
+	FROM information_schema.constraint_column_usage ccu
+    INNER JOIN information_schema.referential_constraints rc
+        ON ccu.constraint_name = rc.constraint_name 
+    INNER JOIN information_schema.key_column_usage kcu 
+        ON kcu.constraint_name = rc.unique_constraint_name  
+	WHERE ccu.table_schema = ? AND ccu.constraint_schema = ? AND ccu.table_name = ?
+	`
+
+	var rows *sql.Rows
+	var err error
+	if rows, err = m.dbConn.Query(query, schema, schema, tableName); err != nil {
+		return nil, err
+	}
+
+	for rows.Next() {
+		var fkey bdb.ForeignKey
+		var sourceTable string
+
+		fkey.Table = tableName
+		err = rows.Scan(&fkey.Name, &sourceTable, &fkey.Column, &fkey.ForeignTable, &fkey.ForeignColumn)
+		if err != nil {
+			return nil, err
+		}
+
+		fkeys = append(fkeys, fkey)
+	}
+
+	if err = rows.Err(); err != nil {
+		return nil, err
+	}
+
+	return fkeys, nil
+}
+
+// TranslateColumnType converts postgres database types to Go types, for example
+// "varchar" to "string" and "bigint" to "int64". It returns this parsed data
+// as a Column object.
+func (m *MSSQLDriver) TranslateColumnType(c bdb.Column) bdb.Column {
+	if c.Nullable {
+		switch c.DBType {
+		case "tinyint":
+			c.Type = "null.Int8"
+		case "smallint":
+			c.Type = "null.Int16"
+		case "mediumint":
+			c.Type = "null.Int32"
+		case "int":
+			c.Type = "null.Int"
+		case "bigint":
+			c.Type = "null.Int64"
+		case "real":
+			c.Type = "null.Float32"
+		case "float":
+			c.Type = "null.Float64"
+		case "boolean", "bool", "bit":
+			c.Type = "null.Bool"
+		case "date", "datetime", "datetime2", "smalldatetime", "time":
+			c.Type = "null.Time"
+		case "binary", "varbinary":
+			c.Type = "null.Bytes"
+		case "timestamp", "rowversion":
+			c.Type = "null.Bytes"
+		case "uniqueidentifier", "xml":
+			c.Type = "null.String"
+		default:
+			c.Type = "null.String"
+		}
+	} else {
+		switch c.DBType {
+		case "tinyint":
+			c.Type = "int8"
+		case "smallint":
+			c.Type = "int16"
+		case "mediumint":
+			c.Type = "int32"
+		case "int":
+			c.Type = "int"
+		case "bigint":
+			c.Type = "int64"
+		case "real":
+			c.Type = "float32"
+		case "float":
+			c.Type = "float64"
+		case "boolean", "bool", "bit":
+			c.Type = "bool"
+		case "date", "datetime", "datetime2", "smalldatetime", "time":
+			c.Type = "time.Time"
+		case "binary", "varbinary":
+			c.Type = "[]byte"
+		case "timestamp", "rowversion":
+			c.Type = "[]byte"
+		case "uniqueidentifier", "xml":
```

Comment:
Looking at this as a result of the above topic, is this supposed to be a null.String? It's clearly in the non-nullable field section.